hw/drivers/sensors/lsm303dlhc/src/lsm303dlhc.c

```diff
@@ -0,0 +1,398 @@
+    };
+
+    /* Register write */
+    payload = reg;
+    rc = hal_i2c_master_write(MYNEWT_VAL(LSM303DLHC_I2CBUS), &data_struct,
+                              OS_TICKS_PER_SEC / 10, 1);
+    if (rc) {
+        LSM303DLHC_ERR("I2C access failed at address 0x%02X\n", addr);
+        #if MYNEWT_VAL(LSM303DLHC_STATS)
+        STATS_INC(g_lsm303dlhcstats, errors);
+        #endif
+        goto error;
+    }
+
+    /* Read one byte back */
+    payload = 0;
+    rc = hal_i2c_master_read(MYNEWT_VAL(LSM303DLHC_I2CBUS), &data_struct,
+                             OS_TICKS_PER_SEC / 10, 1);
+    *value = payload;
+    if (rc) {
+        LSM303DLHC_ERR("Failed to read from 0x%02X:0x%02X\n", addr, reg);
+        #if MYNEWT_VAL(LSM303DLHC_STATS)
+        STATS_INC(g_lsm303dlhcstats, errors);
+        #endif
+    }
+
+error:
+    return rc;
+}
+
+int
+lsm303dlhc_read48(uint8_t addr, uint8_t reg, int16_t *x, int16_t*y, int16_t *z)
+{
+    int rc;
+    uint8_t payload[7] = { reg | 0x80, 0, 0, 0, 0, 0, 0 };
+
+    struct hal_i2c_master_data data_struct = {
+        .address = addr,
+        .len = 1,
+        .buffer = payload
+    };
+
+    /* Register write */
+    rc = hal_i2c_master_write(MYNEWT_VAL(LSM303DLHC_I2CBUS), &data_struct,
+                              OS_TICKS_PER_SEC / 10, 1);
+    if (rc) {
+        LSM303DLHC_ERR("I2C access failed at address 0x%02X\n", addr);
+        #if MYNEWT_VAL(LSM303DLHC_STATS)
+        STATS_INC(g_lsm303dlhcstats, errors);
+        #endif
+        goto error;
+    }
+
+    /* Read six bytes back */
+    memset(payload, 0, sizeof(payload));
+    data_struct.len = 6;
+    rc = hal_i2c_master_read(MYNEWT_VAL(LSM303DLHC_I2CBUS), &data_struct,
+                             OS_TICKS_PER_SEC / 10, 1);
+
+    /* Shift 12-bit left-aligned accel values into 16-bit int */
+    *x = ((int16_t)(payload[0] | (payload[1] << 8))) >> 4;
+    *y = ((int16_t)(payload[2] | (payload[3] << 8))) >> 4;
+    *z = ((int16_t)(payload[4] | (payload[5] << 8))) >> 4;
+
+    if (rc) {
+        LSM303DLHC_ERR("Failed to read from 0x%02X:0x%02X\n", addr, reg);
+        #if MYNEWT_VAL(LSM303DLHC_STATS)
+        STATS_INC(g_lsm303dlhcstats, errors);
+        #endif
+        goto error;
+    }
+
+    /* ToDo: Log raw reads */
+    // console_printf("0x%04X\n", (uint16_t)payload[0] | ((uint16_t)payload[1] << 8));
+
+error:
+    return rc;
+}
+
+/**
+ * Expects to be called back through os_dev_create().
+ *
+ * @param The device object associated with this accellerometer
+ * @param Argument passed to OS device init, unused
+ *
+ * @return 0 on success, non-zero error on failure.
+ */
+int
+lsm303dlhc_init(struct os_dev *dev, void *arg)
+{
+    struct lsm303dlhc *lsm;
+    struct sensor *sensor;
+    int rc;
+
+    lsm = (struct lsm303dlhc *) dev;
+
+    #if MYNEWT_VAL(LSM303DLHC_LOG)
+    log_register("lsm303dlhc", &_log, &log_console_handler, NULL, LOG_SYSLEVEL);
+    #endif
+
+    sensor = &lsm->sensor;
+
+    #if MYNEWT_VAL(LSM303DLHC_STATS)
```

Comment:
Sorry, I originally had them that way and should have scanned the style guide again. I'm always on the fence about this. Having them left-aligned also seriously hurts code readability in my opinion since you lose the flow inside functions, but the norm is more left-aligned. In the future I'll stick to full left-alignment, thanks for the heads up.